### PR TITLE
Pyhton client to connect to running core server

### PR DESCRIPTION
**Pull request by tmbo**
**Proposed changes**:
- Pyhton client to connect to running core server
- an agent that not directly load s a model but rather uploads it to a server and serves it from there

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog

### CODE REVIEW

File: rasa_core/remote.py [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+        while max_retries > 0:
+            with io.open(model_zip, "rb") as f:
+                response = requests.post(url, files={"model": f})
+            max_retries -= 1
+            if response.status_code == 200:
+                logger.debug("Finished uploading")
+                return response.json()
+            else:
+                time.sleep(2)
+                max_retries -= 1
```
[the remark this answers is not in the document]
Comment (ricwo) [reply rasa_core/remote.py]: And we could shave off 2 seconds with
`if max_retries > 0: time.sleep(2)`, if we're in the last iteration of the loop

File: rasa_core/remote.py [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+                logger.error("Rasa Core did not return an action. Response: "
+                             "{}".format(response))
+                response = None
+        logger.info("Done processing message")
+
+    @classmethod
+    def load(cls,
+             path,  # type: Text
+             core_host,  # type: Text
+             auth_token,  # type: Optional[Text]
```
Comment (ricwo) [inline rasa_core/remote.py]: `auth_token=None`

File: rasa_core/agent.py [hunk 3]
```diff
@@ -138,9 +160,13 @@ def toggle_memoization(self, activate):
             if type(p) == MemoizationPolicy:
                 p.toggle(activate)
 
-    def train(self, resource_name=None, model_path=None, remove_duplicates=True,
-              **kwargs):
-        # type: (Optional[Text], Optional[Text], **Any) -> None
+    def train(self,
+              resource_name=None, # type: Optional[Text]
+              model_path=None,    # type: Optional[Text]
+              remove_duplicates=True,   # type: bool
```
[the remark this answers is not in the document]
Comment (tmbo) [reply rasa_core/agent.py]: no, `Optional[bool]` means allowed values are `None, True, False` - it does not mean "this parameter is optional to pass" - but in this case it is either true or false, never `None`

File: rasa_core/remote.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (tmbo) [reply rasa_core/remote.py]: very good catch :+1:

**Pull request merged**